src/python/pants/option/optionable.py

```diff
@@ -6,14 +6,31 @@
                         unicode_literals, with_statement)
 
 import re
+from collections import namedtuple
+from hashlib import sha1
 
 from six import string_types
+from twitter.common.collections import OrderedSet
 
+from pants.option.arg_splitter import GLOBAL_SCOPE
 from pants.option.errors import OptionsError
 from pants.option.scope import ScopeInfo
+from pants.util.memo import memoized_method, memoized_property
 from pants.util.meta import AbstractClass
 
 
+class SubsystemDependency(namedtuple('_SubsystemDependency', ('subsystem_cls', 'scope'))):
+  def subsystem_dependency_joined_scope(self):
+    return self.subsystem_cls.subscope(self.scope)
+
+class SubsystemClientError(Exception): pass
+
+class Register(namedtuple('_Register', ('options', 'optionable', 'bootstrap', 'scope'))):
+  def __call__(self, *args, **kwargs):
+    kwargs['registering_class'] = self.optionable
+    # print('args: {}, kwargs: {}'.format(args, kwargs))
```

Comment:
xx

Reply:
The commented-out code has been removed.